react/components/SliderContext.tsx

```diff
@@ -206,19 +207,49 @@ const SliderContextProvider: FC<SliderContextProps> = ({
 
   const newSlides = preRenderedSlides.concat(slides, postRenderedSlides)
 
-  const slideWidth = useMemo(() => 100 / newSlides.length, [newSlides.length])
+  const slideWidth = useMemo(() => {
+    const baseSlideWidth = 100 / newSlides.length
+
+    let resultingSlideWidth = baseSlideWidth
+
+    if (centerMode !== 'disabled') {
+      resultingSlideWidth =
+        (resolvedSlidesPerPage / (resolvedSlidesPerPage + 1)) * baseSlideWidth
+    }
+
+    return resultingSlideWidth
+  }, [newSlides.length, centerMode, resolvedSlidesPerPage])
 
   const transformMap = useMemo(() => {
     const currentMap: Record<number, number> = {}
 
     newSlides.forEach((_, idx) => {
       const currIdx = infinite ? idx - resolvedSlidesPerPage : idx
+      let transformValue = -(slideWidth * idx)
+
+      if (centerMode !== 'disabled') {
+        // This represents the new value for each transformValue taking into
+        // account the changes made to slideWidth's value due to the fact that
+        // centerMode is enabled.
+        const adjustedTransformValue = -(
+          (1 + 1 / (4 * resolvedSlidesPerPage)) *
+          slideWidth *
+          idx
+        )
+
+        // This is a correction factor to center the slides when centerMode
+        // is enabled and set to 'center'.
+        const transformCenterCorrection =
+          centerMode === 'center' ? (slideWidth * 3) / 8 : 0
+
+        transformValue = adjustedTransformValue + transformCenterCorrection
```

Comment:
What do you think about refactoring it to something like:

```suggestion
transformValue = adjustedTransformValue 

if (centerMode === 'center') {
  transformValue = (slideWidth * 3) / 8
}
```

?

Reply:
I went for the least amount of reassignments to the `transformValue` variable... but I could change to what you've suggested. Do you think it improves readability?

Reply:
I think it separates more clearly what's happening when 👀  But it's up to you

Reply:
Okay! I've just changed it :)